blobstore/blobstore.go

```diff
@@ -197,3 +246,49 @@ $$ LANGUAGE plpgsql`,
 		`CREATE TRIGGER delete_file BEFORE UPDATE OF deleted_at ON files FOR EACH ROW EXECUTE PROCEDURE delete_file()`,
 	)
 }
+
+type command struct {
+	usage     string
+	f         interface{}
+	optsFirst bool
+}
+
+var commands = make(map[string]*command)
+
+func register(cmd string, f interface{}, usage string) *command {
+	switch f.(type) {
+	case func(*docopt.Args) error, func() error, func():
+	default:
+		panic(fmt.Sprintf("invalid command function %s '%T'", cmd, f))
+	}
+	c := &command{usage: strings.TrimLeftFunc(usage, unicode.IsSpace), f: f}
+	commands[cmd] = c
+	return c
+}
+
+func runCommand(name string, args []string) (err error) {
+	argv := make([]string, 1, 1+len(args))
+	argv[0] = name
+	argv = append(argv, args...)
+
+	cmd, ok := commands[name]
+	if !ok {
+		return fmt.Errorf("%s is not a flynn-blobstore command. See 'flynn-blobstore help'", name)
+	}
+	parsedArgs, err := docopt.Parse(cmd.usage, argv, true, "", cmd.optsFirst)
+	if err != nil {
+		return err
+	}
+
+	switch f := cmd.f.(type) {
```

Comment:
All of the registered functions have type `func (args *docopt.Args) error` so there is no need for the type switch.